Setting this magic before the request is required to get VBE 2.0 information,
but also to get all of the supported video modes on some (old?) BIOSes.

See:
http://www.ctyme.com/intr/rb-0273.htm
https://git.savannah.gnu.org/cgit/grub.git/tree/grub-core/video/i386/pc/vbe.c?id=9ee5ae1fae1d7e19f76d4758d9658af90018b0f9#n522

--- linux-5.10.orig/arch/x86/boot/vesa.h
+++ linux-5.10/arch/x86/boot/vesa.h
@@ -25,6 +25,7 @@ struct vesa_general_info {
 } __attribute__ ((packed));
 
 #define VESA_MAGIC ('V' + ('E' << 8) + ('S' << 16) + ('A' << 24))
+#define VBE2_MAGIC ('V' + ('B' << 8) + ('E' << 16) + ('2' << 24))
 
 struct vesa_mode_info {
 	u16 mode_attr;		/* 0 */
--- linux-5.10.orig/arch/x86/boot/video-vesa.c
+++ linux-5.10/arch/x86/boot/video-vesa.c
@@ -38,6 +38,7 @@ static int vesa_probe(void)
 
 	video_vesa.modes = GET_HEAP(struct mode_info, 0);
 
+	vginfo.signature = VBE2_MAGIC;
 	initregs(&ireg);
 	ireg.ax = 0x4f00;
 	ireg.di = (size_t)&vginfo;
